persistence/src/main/java/de/cyface/persistence/content/MeasurementProviderHelper.kt

```diff
@@ -0,0 +1,285 @@
+        identifierTable = IdentifierTable(context)
+    }
+
+    /**
+     * Deletes one or multiple rows (depending on the format of the provided URI) from the database. If you delete a
+     * [de.cyface.persistence.model.Measurement] all data linked via `ForeignKey` is cascadingly deleted as well.
+     *
+     * @param uri The URI specifying the table to delete from. If this ends with a single numeric identifier that row is
+     * deleted otherwise multiple rows might be deleted depending on the `selection` and
+     * `selectionArgs`.
+     * @param selection The part of an SQL where statement behind the where. You can use '?' as a placeholder to secure
+     * yourself from SQL injections.
+     * @param selectionArgs The arguments to place inside the '?' placeholder from `selection`.
+     * @return The number of rows deleted.
+     */
+    fun deleteRow(
+        uri: Uri,
+        selection: String?,
+        selectionArgs: Array<String>?
+    ): Int {
+        val pathSegments = uri.pathSegments
+        val database = writableDatabase
+
+        val table: CyfaceTable = matchTable(uri)
+
+        var ret = 0
+        database.beginTransaction()
+        return try {
+            when (pathSegments.size) {
+                2 -> {
+                    val rowIdentifier = pathSegments[1]
+                    when (pathSegments[0]) {
+                        // FIXME: ensure the measurement-related data is deleted automatically cascadingly (write/execute test)
+                        MeasurementTable.URI_PATH, LocationTable.URI_PATH, PressureTable.URI_PATH, EventTable.URI_PATH -> {
+                            val adaptedSelection = (BaseColumns.ID + "=" + rowIdentifier
+                                    + if (selection == null) "" else " AND $selection")
+                            ret += table.deleteRow(database, adaptedSelection, selectionArgs)
+                            database.setTransactionSuccessful()
+                            ret
+                        }
+                        else -> throw IllegalStateException("Unknown table identified by content provider URI: $uri")
+                    }
+                }
+                1 -> {
+                    when (pathSegments[0]) {
+                        // FIXME: ensure the measurement-related data is deleted automatically cascadingly (write/execute test)
+                        MeasurementTable.URI_PATH, LocationTable.URI_PATH, PressureTable.URI_PATH, EventTable.URI_PATH, IdentifierTable.URI_PATH -> {
+                            ret += table.deleteRow(database, selection, selectionArgs)
+                            database.setTransactionSuccessful()
+                            ret
+                        }
+                        else -> throw IllegalStateException(
+                            "Unable to delete data from table corresponding to URI $uri. There seems to be no such table."
+                        )
+                    }
+                }
```

Comment:
Before this was:

```
switch (pathSegments.get(0)) {
    case MeasurementTable.URI_PATH:
        // Measurement requires to also delete all dependent entries and then call table.deleteRow
        // All other database entries just call table.deleteRow directly.

        ret += deleteDataForMeasurement(database, Long.parseLong(rowIdentifier));
        // continues here until return ! -->
    case GeoLocationsTable.URI_PATH:
    case EventTable.URI_PATH:
        // Add the id specified by the URI to implement expected behaviour of a content resolver, where
        // the last element of the path is the identifier of the element requested. This is only
        // necessary for single row deletions.
        String adaptedSelection = BaseColumns._ID + "=" + rowIdentifier
                + (selection == null ? "" : " AND " + selection);
        ret += table.deleteRow(getWritableDatabase(), adaptedSelection, selectionArgs);
        database.setTransactionSuccessful();
        return ret;
    default:
        throw new IllegalStateException("Unknown table identified by content provider URI: " + uri);
}
```

I was able to simplify this as the measurement-related data is now automatically deleted cascadingly via `ForeignKey`.